docs/source/write_first_app.rst

```diff
@@ -164,329 +188,673 @@ You should see the following:
+      const asset = JSON.parse(assetString);
+      asset.Owner = newOwner;
+      return ctx.stub.putState(id, Buffer.from(JSON.stringify(asset)));
+  }
+   // ReadAsset returns the asset stored in the world state with given id.
+  async ReadAsset(ctx, id) {
+      const assetJSON = await ctx.stub.getState(id); // get the asset from chaincode state
+      if (!assetJSON || assetJSON.length === 0) {
+          throw new Error(`The asset ${id} does not exist`);
+      }
+      return assetJSON.toString();
+  }
 
-Save the program and navigate back to your ``fabcar/javascript`` directory.
-Now run the ``query`` program again:
+Terminal output:
 
 .. code:: bash
 
-  node query.js
+  ***********************
+  Submit Transaction: TransferAsset asset1 from owner Tomoko > owner Tom
+  Evaluate Transaction: ReadAsset asset1
+    result: {
+    "ID": "asset1",
+    "Color": "blue",
+    "Size": "5",
+    "Owner": "Tom",
+    "AppraisedValue": "350"
+  }
 
-You should see the following:
 
-.. code:: json
+A closer look
+-------------
 
-  Wallet path: ...fabric-samples/fabcar/javascript/wallet
-  Transaction has been evaluated, result is:
-  {"color":"black","docType":"car","make":"Tesla","model":"S","owner":"Adriana"}
+Let's take a closer look at how the sample javascript application uses the APIs provided by the
+`Fabric Node SDK <https://hyperledger.github.io/fabric-sdk-node/>`__ to
+interact with our Fabric network. Use an editor (e.g. atom or visual studio) to
+open ``app.js`` located in the ``asset-transfer-basic/application-javascript`` directory.
 
-If you go back and look at the result from when the transaction was
-``queryAllCars``, you can see that ``CAR4`` was Adriana’s black Tesla model S,
-which is the result that was returned here.
+The application starts by bringing in scope two key classes from the
+``fabric-network`` module; ``Wallets`` and ``Gateway``. These classes
+will be used to locate the ``appUser`` identity in the wallet, and use it to
+connect to the network:
 
-We can use the ``queryCar`` transaction to query against any car, using its
-key (e.g. ``CAR0``) and get whatever make, model, color, and owner correspond to
-that car.
+.. code:: bash
 
-Great. At this point you should be comfortable with the basic query transactions
-in the smart contract and the handful of parameters in the query program.
+  const { Gateway, Wallets } = require('fabric-network');
 
-Time to update the ledger...
+First, the program uses the Wallet class to get our application user from our file system.
 
-Updating the ledger
--------------------
+.. code:: bash
 
-Now that we’ve done a few ledger queries and added a bit of code, we’re ready to
-update the ledger. There are a lot of potential updates we could make, but
-let's start by creating a **new** car.
+  const identity = await wallet.get(registerUser.ApplicationUserId);
 
-From an application perspective, updating the ledger is simple. An application
```

Comment:
I think this para about Update to the ledger got lost in the new description. Seem like a good detail to keep along with the tutorial/write_first_app.diagram.2.png diagram

Reply:
Agreed, re-added paragraph and image.